### PR TITLE
Add driver for HDC1000 Humidity/Temperature Sensor.

### PR DESCRIPTION
**Pull request by wurmmi**
Add a kernel driver module that interacts with an FPGA core, which provides data from the HDC1000 Humidity/Temperature Sensor.

### CODE REVIEW

File: drivers/sevensegment/sevensegment.c [hunk 1]
```diff
@@ -77,11 +78,16 @@ static int sevenseg_write(struct file *filep, const char *buf,
...
 
+  /* limit number of writeable bytes to maximum which is still possible */
   if ((*offp + count) > BUF_SIZE)
     count = BUF_SIZE - *offp;
 
+  /* copy data from user space into kernel space buffer */
   if (count > 0)
+  {
     count = count - copy_from_user(sevenseg->buffer + *offp, buf, count);
+    *offp += count;
```
Comment (wurmmi) [inline drivers/sevensegment/sevensegment.c]: @skaupper dieses offp incrementieren gehört hier hinein, oder?

Comment (skaupper) [reply drivers/sevensegment/sevensegment.c]: sollte keinen unterschied machen, oder? unten wo dus entfernt hast wäre count sowieso auf 0

File: drivers/humid_temp/humid_temp.c [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  if (count > 0)
+  {
+    count = count - copy_from_user(dev->buffer + *offp, buf, count);
+    *offp += count;
+  }
+
+  /*
+   * There is probably nothing to write in this driver..
+   * We might consider to remove the entire humid_temp_write() function.
+   */
```
Comment (skaupper) [inline drivers/humid_temp/humid_temp.c]: würd auch sagen, dass wir die write einfach rausnehmen. wir haben nix zu schreiben und der kernel unterstützt eh character devices ohne write

File: drivers/sevensegment/sevensegment.c (hunk 1 above)
Comment (wurmmi) [reply drivers/sevensegment/sevensegment.c]: stimmt.. thanks 👍

**Comment by wurmmi** [comment]
Completed implementation from my side.
Waiting on @pagdot to provide an FPGA image to test with.

Once the full integration (FPGA + driver + user space) was tested, we can merge this PR.

**Comment by wurmmi** [comment]
Integration seems to work so far:

Screenshot from 2019-11-25 23-01-09

See dashboard at:
http://193.170.192.224:15000/d/m4azNIbZz/sensor_panels

**Next steps:**
- wait on progress in #5 to read actual sensor data
- test again
- merge
- change repository pointers in [yocto repo here](https://github.com/skaupper/SmartSystemsLab-Yocto/blob/master/layer/meta-application/recipes-ssl/humid-temp-mod/humid-temp-mod_1.0.bb#L7)

**Comment by wurmmi** [comment]
Testing shows that FPGA HDL part is not working.

Thus - merging this PR now, in order to be able to continue working for Milestone 3.

**Pull request merged**